cmake/googletest.cmake

```diff
@@ -0,0 +1,17 @@
+# Google Test library
+#
+# Google Test is a testing framework for C++ developed by Google.
+#
+# This CMake script integrates Google Test into the project, allowing easy usage.
+# Usage example: target_link_libraries(my_target PRIVATE GTest::GTest)
+
+include(FetchContent)
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.12.1
+)
+FetchContent_MakeAvailable(googletest)
+add_library(GTest::GTest INTERFACE IMPORTED)
+target_link_libraries(GTest::GTest INTERFACE gtest_main)
```

Comment:
I believe that these two commands (add_library & target_link_libraries) are not required. All you need to do, is to link against `GTest::gtest_main` in the next commit.

Reply:
Updated